File: problem_228/sol.py
class Solution(object):
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """

        begin = ""
        end = ""
        sum_range = []
        for num in nums:
            if begin == "":
                begin = num
                tmp = num + 1
            elif tmp == num:
                end = num
                tmp += 1
            else:
                if end == "":
                    sum_range.append("%d" % begin)
                else:
                    sum_range.append("%d->%d" % (begin,end))
                begin = num
                tmp = num + 1
                end = ""
        if num == nums[len(nums)-1]:
            if end == "":
                sum_range.append("%d" % begin)
            else:
                sum_range.append("%d->%d" % (begin,end))
        return sum_range

File: 283_Move_Zeroes/sol.cpp
class Solution {
    public:
        void moveZeroes(vector<int>& nums) {
            int num_of_non_zero = 0;
            for(int i=0;i<nums.size();i++)
                if(nums[i] != 0)
                    nums[num_of_non_zero++] = nums[i];
            for(int i=num_of_non_zero;i<nums.size();i++)
                nums[i] = 0;
        }
};
